### PR TITLE
fix(es/minifier): Abort inliner on fn declarations used multiple time

### PR DESCRIPTION
**Pull request by kdy1**
**Description:**

**Related issue:**

 - Closes https://github.com/swc-project/swc/issues/6463.

### CODE REVIEW

**Review by kdy1 — COMMENTED** [review]
swc-bump:
 - swc_ecma_minifier

File: crates/swc_ecma_minifier/tests/fixture/issues/6463/output.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+var foo_1 = function() {
```
Comment (jridgewell) [inline crates/swc_ecma_minifier/tests/fixture/issues/6463/output.js:1]: Is it not possible to avoid the reassignment?

**Review by jridgewell — APPROVED** [review]

File: crates/swc_ecma_minifier/tests/fixture/issues/6463/output.js (hunk 1 above)
Comment (kdy1) [reply crates/swc_ecma_minifier/tests/fixture/issues/6463/output.js:1]: I think it's possible. I'll check, thank you!

Comment (kdy1) [reply crates/swc_ecma_minifier/tests/fixture/issues/6463/output.js:1]: I tried, but we can't detect it while deciding to inline a function declaration. We need a costly reverse map to access `foo` from `bar = foo`.

I'll revisit this in the future

**Pull request merged**